### ISSUE TITLE
Can't reformat correctly in a round with `trailing_comma_on_call_site` enabled

### ISSUE THREAD

**Issue by Goooler** [pinterest/ktlint#1642]
## Expected Behavior

Reformat correctly in a round with `trailing_comma_on_call_site` enabled.

## Observed Behavior
Need to format correctly code below for 3 times.

Original:
```kotlin
interface Foo {
  fun foo()
}

fun bar(foo: Foo) {
  foo.foo()
}

fun main() {
  bar(object : Foo {
      override fun foo() {
        TODO("Not yet implemented")
      }
  })
}
```

## Steps to Reproduce
First round reformat:
```kotlin
fun main() {
  bar(object : Foo {
    override fun foo() {
      TODO("Not yet implemented")
    }
  },)
}
```

Second round reformat:
```kotlin
fun main() {
  bar(object : Foo {
    override fun foo() {
      TODO("Not yet implemented")
    }
  },
  )
}
```

Third round:
```kotlin
fun main() {
  bar(
    object : Foo {
      override fun foo() {
        TODO("Not yet implemented")
      }
    },
  )
}
```
You can start from the second step, also need 2 rounds to format.

## Your Environment

* Version of ktlint used: 0.47.1
* Relevant parts of the `.editorconfig` settings https://github.com/Goooler/DemoApp/blob/trunk/.editorconfig
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): CLI
* Version of Gradle used (if applicable): Nope
* Operating System and version: MacOS 12.5.1

**Comment by vanniktech** [comment]
Yep. Noticed this too. For instance here: https://github.com/vanniktech/gradle-code-quality-tools-plugin/pull/245/files#diff-7885ec759a167c69428552655473fc77195bb3733559c7e27bb3b2d26d36452aL21

Note though that I'm stuck on the second loop and it won't do the third and place `object : Instantiator {` into it's own line

**Comment by paul-dingemans** [comment]
In the examples above, the trailing comma should not have been added. Currently it is added because the VALUE_ARGUMENT_LIST is a multiline statement and any newline inside the VALUE_ARGUMENT_LIST is considered to be a reason to add the trailing comma.
